pkg/controller/handlers/mcpcatalog/mcpcatalog.go

```diff
@@ -362,8 +362,7 @@ func (h *Handler) DeleteUnauthorizedMCPServerInstances(req router.Request, _ rou
 
 	// Iterate through each MCPServerInstance and make sure it is still allowed to exist.
 	for _, instance := range mcpServerInstances.Items {
-		if !instance.DeletionTimestamp.IsZero() {
-			// This server is already being deleted, so we don't have to delete it again.
+		if instance.Spec.MCPCatalogName != system.DefaultCatalog || !instance.DeletionTimestamp.IsZero() {
 			continue
 		}
 
```

Comment:
I can only comment on this line because it's part of the diff, but I believe lines 369 through 381 can be removed now, correct?

Reply:
Yes you are correct, thanks

Reply:
fixed